### PR TITLE
feat: Add subnet and virtual network name validation

### LINKED ISSUE #268: Validate subnet resource
- Check that the CIDR block is valid
- Check that the CIDR block is inside the Virtual network
- Check that the name matches the restrictions in https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules
- Check that the availability zone is valid in AWS

Terraform provider issue: https://github.com/multycloud/terraform-provider-multy/issues/55

### PR DESCRIPTION
**Pull request by szymonnogiec**
Resolves #268 issue

### CODE REVIEW

File: resources/types/subnet.go [hunk 1]
```diff
@@ -49,9 +50,13 @@ func NewSubnet(s *Subnet, resourceId string, subnet *resourcespb.SubnetArgs, oth
 }
 
 func (r *Subnet) Validate(ctx resources.MultyContext) (errs []validate.ValidationError) {
-	//if vn.Name contains not letters,numbers,_,- { return false }
-	//if vn.Name length? { return false }
-	//if vn.AvailbilityZone valid { return false }
+	//if vn.AvailbilityZone valid { return false }i
```
Comment (goncalo-rodrigues) [inline resources/types/subnet.go:53]: since you're here, can you please remove this comment? it's no longer accurate as we removed av zones from subnets

**Review by goncalo-rodrigues — CHANGES_REQUESTED** [review]

File: resources/types/virtual_network.go [hunk 2]
```diff
@@ -50,9 +52,13 @@ func NewVirtualNetwork(r *VirtualNetwork, resourceId string, vn *resourcespb.Vir
 
 func (r *VirtualNetwork) Validate(ctx resources.MultyContext) (errs []validate.ValidationError) {
 	errs = append(errs, r.ResourceWithId.Validate()...)
-	//if r.Name contains not letters,numbers,_,- { return false }
-	//if r.Name length? { return false }
 	//if r.CidrBlock valid CIDR { return false }
```
Comment (szymonnogiec) [inline resources/types/virtual_network.go:55]: @goncalo-rodrigues Should I remove this line also? This seams to be implemented.

File: resources/types/subnet.go (hunk 1 above)
Comment (szymonnogiec) [reply resources/types/subnet.go:53]: fixed;

File: resources/types/virtual_network.go (hunk 2 above)
Comment (goncalo-rodrigues) [reply resources/types/virtual_network.go:55]: Oh right, missed that one. Yap!

**Review by goncalo-rodrigues — APPROVED** [review]

**Pull request merged**